pyuvdata/uvdata/mir_meta_data.py

```diff
@@ -0,0 +1,3087 @@
+        discard_flagged : bool
+            If set to True, exclude from metadata where the internal mask has been set
+            to False. Default is False. Note that this cannot be used if setting
+            `merge=True`.
+
+        Returns
+        -------
+        new_obj : MirMetaData object
+            The resultant combination of the two objects.
+        """
+        return self.__add__(
+            other,
+            inplace=True,
+            merge=merge,
+            overwrite=overwrite,
+            discard_flagged=discard_flagged,
+        )
+
+    def copy(self, skip_data=False):
+        """
+        Make and return a copy of the MirMetaData object.
+
+        Parameters
+        ----------
+        skip_data : bool
+            If set to True, forgo copying the data-related attributes. Default is False.
+
+        Returns
+        -------
+        new_obj : MirMetaData object
+            Copy of the original object.
+        """
+        # Initialize a new object of the given type
+        copy_obj = type(self)()
+
+        deepcopy_list = ["_stored_values"]
+        data_list = ["_stored_values", "_data", "_mask", "_header_key_index_dict"]
+
+        for attr in vars(self):
+            if skip_data and attr in data_list:
+                continue
+
+            if attr in deepcopy_list:
+                copy_attr = copy.deepcopy(getattr(self, attr))
+            else:
+                try:
+                    copy_attr = getattr(self, attr).copy()
+                except AttributeError:
+                    copy_attr = copy.deepcopy(getattr(self, attr))
+
+            setattr(copy_obj, attr, copy_attr)
+
+        return copy_obj
+
+    def where(
+        self,
+        select_field,
+        select_comp,
+        select_val,
+        mask=None,
+        return_header_keys=False,
+    ):
+        """
+        Find where metadata match a given set of selection criteria.
+
+        This method will produce a masking screen based on the arguments provided to
+        determine which entries match a given set of conditions.
+
+        Parameters
+        ----------
+        select_field : str
+            Field in the metadata to evaluate.
+        select_comp : str
+            Specifies the type of comparison to do between the value supplied in
+            `select_val` and the metadata. No default, allowed values include:
+            "eq" | "==" (equal to, matching any in `select_val`),
+            "ne" | "!=" (not equal to, not matching any in `select_val`),
+            "lt" | "<" (less than `select_val`),
+            "le" | "<=" (less than or equal to `select_val`),
+            "gt" | ">" (greater than `select_val`),
+            "ge" | ">=" (greater than or equal to `select_val`),
+            "between" (between the range given by two values in `select_val`),
+            "outside" (outside of the range give by two values in `select_val`).
```

Comment:
I'm worried this won't render well in automatically built docs. Speaking of docs, have you added this file to the docs? I think we should (and then we can check if they're rendered as we'd like).

Reply:
So `MirParser` was added, but `MirMetaData` was not, so that's been added in f8f499f0. I did modify this a little bit to be a bit clearer in a2f46b4b -- I think it renders okay by my eye with the more compact text that I put in.